Backend/lib/beckn-services/src/AWS/S3/Flow.hs

```diff
@@ -243,3 +247,44 @@ mockDelete baseDirectory bucketName path =
 
 getFullPathMock :: String -> Text -> String -> String
 getFullPathMock baseDirectory bucketName path = baseDirectory <> "/" <> cs bucketName <> "/" <> path
+
+-- | Generate a pre-signed URL for uploading a video
+generateUploadUrl' ::
+  ( MonadFlow m
+  ) =>
+  Text ->
+  String ->
+  Seconds ->
+  m Text
+generateUploadUrl' = generateUrl' PUT
+
+-- | Generate a pre-signed URL for downloading a video
+generateDownloadUrl' ::
+  ( MonadFlow m
+  ) =>
+  Text ->
+  String ->
+  Seconds ->
+  m Text
+generateDownloadUrl' = generateUrl' GET
+
+data Method = GET | PUT
+
+generateUrl' ::
+  ( MonadFlow m
+  ) =>
+  Method ->
+  Text ->
+  String ->
+  Seconds ->
+  m Text
+generateUrl' method bucketName path expires = withLogTag "S3" $ do
+  env <- Amazonka.newEnv Amazonka.discover
```

Comment:
`amazonka` have standard approach for storing api keys, the same was used for email and slack messages (`sendEmail`, `publishMessage` functions)

> 1. Environment Variables
> Set the following environment variables:
> AWS_ACCESS_KEY_ID
> AWS_SECRET_ACCESS_KEY
> (optionally) AWS_SESSION_TOKEN for temporary credentials
> Amazonka will automatically pick these up when you use AWS.discover or AWS.newEnv AWS.discover.
> 
> 2. AWS Credentials File
> Store your credentials in ~/.aws/credentials (the default AWS CLI location), for example:
> ini
> [default]
> aws_access_key_id = YOUR_ACCESS_KEY
> aws_secret_access_key = YOUR_SECRET_KEY
> 
> Amazonka will use this file if present.